grammar.js

```diff
@@ -18,10 +18,10 @@ module.exports = grammar({
       ),
 
     _default_prompt_line: ($) =>
-      seq(alias(/iex(\(\d+\))?>/, $.prompt), optional($.expression)),
+      seq(alias(/iex(\([^\)]+\)\d*)?>/, $.prompt), optional($.expression)),
```

Comment:
Please try this test case:

```
iex(1)> abs(1)>1
false
```

Pretty sure it will fail and can be solved by non-greedy match `+` -> `+?`

Reply:
Actually the AST would be correct, but I'd make this a highlighting test so we can assert more precisely :D

Reply:
actually I don't think we can do highlight tests because the grammar doesn't have a `$.comment` node :sweat_smile:

maybe it makes sense to add one and just not highlight it? seems like it might get confusing whether this is a result or a comment:

```
iex> self()
#PID<..>
```

Reply:
actually I think if `comment` was under `result` in the tree, that'd work out fine and the injections would just need `(#set! injection.include-children)` iirc

Reply:
I see, here's another shot:

```
iex(4)> #(4)>
```

The AST should include an expression, and I believe with greedy match it wouldn't have one?

Reply:
hmm looks like that case passes too (and the tokens look good in the playground/highlights)

I think I see what you're saying about the greedy matches and now I'm curious why it isn't mis-parsing these too :thinking:

Reply:
Gah nevermind! You did `[^\)]` so it takes care of it, but now we have the test ^^